app/controllers/teachers_controller.rb

```diff
@@ -126,13 +115,12 @@ def delete
 
   private
 
-  def school_from_params
-    School.find_by(name: school_params[:name], city: school_params[:city], state: school_params[:state]) || School.new(school_params)
+  def deny_access
```

Comment:
It may be prudent to make a more specific function name to indicate it's only denying based on email, in the event that in the future you implement another reason to deny access for whatever reason. From an outside perspective, deny_access sounds like an all-encompassing function rather than just an email validation access issue.